futures-core/src/future/mod.rs

```diff
@@ -141,6 +141,15 @@ if_std! {
         }
     }
 
+    #[cfg(feature = "nightly")]
+    impl<F: ?Sized + Future> Future for ::pin_api::PinBox<F> {
```

Comment:
@withoutboats Shouldn't this be `T: StableFuture`?

Reply:
No. StableFuture isn't object safe, so `StableFuture::pin` returns a `PinBox<Future>`, using unsafe code.

Reply:
Gotcha. Thanks!

Reply:
If it were able to be StableFuture, it could just live in futures-stable and there'd be no need for this to be in-tree other than convenience.